vllm/v1/spec_decode/eagle_mixin.py

```diff
@@ -0,0 +1,45 @@
+import torch
+from typing import Optional
+
+class EagleMixin:
+    def propose_chain(self, hidden_states: torch.Tensor) -> torch.Tensor:
```

Comment:
`propose_chain` feels like a poor name for this function, as it conflicts with `EagleProposer.propose()`. How about something closer to "get_draft_logits" or similar?

Reply:
Or "greedy_sample_draft_tokens" might be more appropriate since I now see that this is a shortcut for compute_logits + greedy sampling

Reply:
In what way is `propose_chain` here analogous to `propose_tree`? The latter seems like a much more comprehensive `propose()`-style function, whereas `propose_chain` here seems like a much simpler combination of `compute_logits` and greedy sampling?

Reply:
I see what you mean. I think I would still prefer something like `sample_chain` or `propose_sample_greedy` or similar, but I am okay to go with `propose_chain` if you think it is the best choice.